E2E/tests/voting.test.ts

```diff
@@ -0,0 +1,189 @@
+    game.vote(chessPiece.pawn, chessTile.a1);
+    game.voteIsVisible(chessPiece.pawn, chessTile.a1, chessPiece.pawnValue);
+    I.waitForElement(game.locator.playersList.playerDoneIcon(username.user2));
+    game.vote(chessPiece.knight, chessTile.e2);
+    game.voteIsNotVisible(chessPiece.pawn, chessTile.a1, chessPiece.pawnValue);
+    game.voteIsVisible(chessPiece.knight, chessTile.e2, chessPiece.knightValue);
+  });
+});
+
+Scenario('Players can’t see each others votes until all of them make a move', async ({ I, login, username, roles, game, chessPiece, chessTile }) => {
+  roles.voterLogin();
+  I.waitForText('Waiting for more players');
+  let url = await I.grabCurrentUrl();
+  session(username.user2, () => {
+    login.loginIntoCreatedGameRoom(url, username.user2);
+    roles.voterLogin();
+    I.waitForText('Game in progress');
+    game.vote(chessPiece.knight, chessTile.e4);
+    game.voteIsVisible(chessPiece.knight, chessTile.e4, chessPiece.knightValue);
+  });
+  I.waitForElement(game.locator.playersList.playerDoneIcon(username.user2));
+  game.voteIsNotVisible(chessPiece.knight, chessTile.e4, chessPiece.knightValue);
+});
+
+Scenario('Two users with same name vote as separate players', async ({ I, login, username, roles, game, chessPiece, chessTile }) => {
+  roles.voterLogin();
+  I.waitForText('Waiting for more players');
+  let url = await I.grabCurrentUrl();
+  session(username.user2, () => {
+    login.loginIntoCreatedGameRoom(url, username.user1);
+    roles.voterLogin();
+    I.waitForText('Game in progress');
+    game.vote(chessPiece.knight, chessTile.e4);
+    game.voteIsVisible(chessPiece.knight, chessTile.e4, chessPiece.knightValue);
+  });
+  I.waitForElement(game.locator.playersList.playerDoneIcon(username.user1));
+  game.voteIsNotVisible(chessPiece.knight, chessTile.e4, chessPiece.knightValue);
+  game.vote(chessPiece.queen, chessTile.f5);
+  I.waitForText('Game complete');
+  game.voteIsVisible(chessPiece.queen, chessTile.f5, chessPiece.queenValue);
+  game.voteIsVisible(chessPiece.knight, chessTile.e4, chessPiece.knightValue);
+  I.see(username.user1, game.locator.playersList.firstUserInList);
+  I.see(username.user1, game.locator.playersList.secondUserInList);
+  session(username.user2, () => {
+    game.voteIsVisible(chessPiece.queen, chessTile.f5, chessPiece.queenValue);
+    game.voteIsVisible(chessPiece.knight, chessTile.e4, chessPiece.knightValue);
+    I.see(username.user1, game.locator.playersList.firstUserInList);
+    I.see(username.user1, game.locator.playersList.secondUserInList);
+  });
+});
+
+Scenario('3 players place different figures on different board squares', async ({ I, login, username, roles, game, chessPiece, chessTile }) => {
+  roles.voterLogin();
+  I.waitForText('Waiting for more players');
+  let url = await I.grabCurrentUrl();
+  session(username.user2, () => {
+    login.loginIntoCreatedGameRoom(url, username.user2);
+    roles.voterLogin();
+    I.waitForText('Game in progress');
```

Comment:
This code seems to repeat quite a lot, if you decide to go with the User class I described below, it would make sense to move it there.